File: src/main/java/com/tram/springbootangularboard/domain/AccountRepository.java
package com.tram.springbootangularboard.domain;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.CrudRepository;

import java.util.Optional;

public interface AccountRepository extends JpaRepository<Account, Long> {
    Optional<Account> findBySocialId(String socialId);
    Optional<Account> findByEmail(String email);
}

File: src/main/java/com/tram/springbootangularboard/domain/AccountRepository.kt
package com.tram.springbootangularboard.domain

import org.springframework.data.repository.CrudRepository
import java.util.*

interface AccountRepository : CrudRepository<Account, Long> {
    fun findByUserId(userId: String): Optional<Account>?
    fun findBySocialId(socialId: String): Optional<Account>?
}